### PR TITLE
feat: add segment and node lookups

### PR DESCRIPTION
**Pull request by tchardin**
Would love to use this package for a HAMT reifier. Will add another PR here with the actual Reifier at some point.

The aim of the reifier is to fetch partial Filecoin state from Lotus nodes via GraphSync.

### CODE REVIEW

**Comment by willscott** [comment]
see if https://github.com/ipld/go-ipld-adl-hamt/pull/33 still works for the reify part of what you want :)

**Comment by tchardin** [comment]
Yup, just had to switch it to bindnode types!

File: node.go [hunk 1]
```diff
@@ -86,12 +86,16 @@ func (n *Node) LookupByString(s string) (ipld.Node, error) {
 	return n.lookupValue(&n.Hamt, n.bitWidth(), 0, hk, key)
 }
 
-func (*Node) LookupByNode(ipld.Node) (ipld.Node, error) {
-	panic("TODO")
+func (n *Node) LookupByNode(key ipld.Node) (ipld.Node, error) {
+	ks, err := key.AsString()
```
Comment (willscott) [inline node.go:90]: Will keys always be a string?
Wonder if it makes sense to also handle cases where the type is Bytes?

**Review by willscott — APPROVED** [review]

File: node.go (hunk 1 above)
Comment (tchardin) [reply node.go:90]: Ah, you're right, I assumed same interface as the Filecoin HAMT which always passes keys as strings but we could support bytes too for more flexibility.
Also do you think I could get added to the IPLD org? Would love to help maintain and contribute as much as I can.

Comment (willscott) [reply node.go:90]: > do you think I could get added to the IPLD org?

that part is self-serve now. Propose adding yourself with a PR modifying https://github.com/ipld/github-mgmt/blob/master/github/ipld.yml

Comment (tchardin) [reply node.go:90]: Thanks!

**Pull request merged**